### PR TITLE
Support reading ClientHello in QUIC initial packet

### PR DESCRIPTION
**Pull request by cuonglm**

### CODE REVIEW

**Comment by alexelisenko** [comment]
@cuonglm Looks good, but lets move https://github.com/cuonglm/quicsni to the Windscribe github.

**Comment by cuonglm** [comment]
> @cuonglm Looks good, but lets move https://github.com/cuonglm/quicsni to the Windscribe github.

Done.

File: go.mod [hunk 1]
```diff
@@ -2,19 +2,24 @@ module github.com/Windscribe/go-vhost
 
 go 1.19
 
-require github.com/quic-go/quic-go v0.34.0
+require (
+	github.com/cuonglm/quicsni v0.0.1
```
Comment (alexelisenko) [inline go.mod]: should reference the Windscribe repo

Comment (cuonglm) [reply go.mod]: See the `replace` directive below.

Comment (alexelisenko) [reply go.mod]: whats the benefit of using a replace here?

Comment (cuonglm) [reply go.mod]: This is how go modules works. The modules name is not replaced, but "where" to fetch the modules. The `replace` tells `go` whenever doing something with module `github.com/cuonglm/quicsni`, go to `github.con/Windscribe/quicsni` and fetch the module content there.

If you do really want to refer to Windscribe only, we can rename the module in Windscribe repo and use it as import path here.

Comment (cuonglm) [reply go.mod]: Anyway, I changed it to use Windscribe/quicsni, PTAL!

**Pull request merged**